File: assignments/three/GasolineCar.py
from Car import Car

class GasolineCar (Car):
    def __init__(self, makeAndModel="none assigned", numberOfDoors=4, maximumNumberOfPassengers=5, gasTankSize=-1):
        super().__init__(makeAndModel,numberOfDoors,maximumNumberOfPassengers)
        self.gasTankSize = gasTankSize
    def setGasTankSize(self,newGasTankSize:int):
        self.gasTankSize = newGasTankSize
    def getGasTankSize(self)->int:
        return self.gasTankSize
    def __str__(self):
        s = super().__str__()
        s += ", Gas Tank Size: "+ str(self.getGasTankSize())
        return s
    
if __name__=="__main__":
    c1 = GasolineCar("Toyota Corolla 2000", 4,6,8)
    print(c1)
    
        

File: assignments/four/GasolineCar.java

public class GasolineCar extends Car{
    private double gasTankSize;
    public GasolineCar(String makeAndModel,int maximumNumberOfPassangers, int numberOfDoors,double gasTankSize){
        super(makeAndModel,maximumNumberOfPassangers,numberOfDoors);
        this.gasTankSize = gasTankSize;
    }
    public double getGasTankSize(){
        return gasTankSize;
    }
    public void setGasTankSize(double g){
        this.gasTankSize = g;
    }
    public String toString(){
        return super.toString() + "\nGas Tank Size: " + gasTankSize;
    }
}
/*
GasolineCar: (11 points total)
GasolineCar(String makeAndModel, int maximumNumberOfPassengers, int numberOfDoors,
          double gasTankSize 5 points
         toString()→ string 3 points
setGasTankSize(double g) 2 points
getGasTankSize()→ double 1 point
 */